### DISCUSSION TITLE
A question about taiko's SV conversion

### DISCUSSION THREAD

**Discussion by msmcculley** [ppy/osu#22734] [Q&A]
Why is it that the spinners are coming in slowly (and obstructively, if I may add) when the SV is slow enough, but other slow SV parts seem to not do the same obstruction, is it simply because thats how the game is loading them in? 

Essentially this behavior makes the map *really* difficult to play and is genuinely just not fun to interact with, apart from stable's playable way, which actually reverses the scroll speed and has them coming in super fast, not slow.

Not sure if this is under the "skill issue" category or if this could be discussed?

Video for context:

https://user-images.githubusercontent.com/90729492/221341463-f5d4ee10-55fd-4bf1-bcbb-34f7fd313698.mp4

**Comment by Axilotl17** [comment]
Ive looked over a little bit of this and i have this insight too add: 

map: https://osu.ppy.sh/beatmapsets/1598413#taiko/3264362

first instance of spinner (or denden note) is at `00:36:182`:

- relevant timing point: `36182,-10000,4,1,0,75,0,0`
- relevant hit object: `256,192,36182,12,2,37441,0:0:0:0:`

it appears that the `beatLength` of the timing point is `-10000`, and according to the [relevant osu! wiki page](https://osu.ppy.sh/wiki/en/Client/File_formats/Osu_%28file_format%29#timing-points) this is treated as a percent, and the equation for the spinner scroll speed velocity *should be* something like `SliderMultiplier/(-0.01*beatLength)`

> a negative inverse slider velocity multiplier, as a percentage. For example, -50 would make all sliders in this timing section twice as fast as SliderMultiplier.

as the `SliderMultipler` is `1.4` and `beatLength` is `-10000` we should get `1.4/(-0.01*-10000)` which simplifies to `0.014`. this would result in a very slow scroll speed, as is demonstrated by the video. 
for context, the note at `00:17:721` has a scroll speed of `2.8`, which is about 200 times faster, which sounds about right for the speed in gameplay for LAZER.

however, this behavior is NOT replicated in stable. it appears these notes teleport in with no warning given, with a practically infinite scroll speed. 
I agree that the lazer implementation follows the wiki, but i also can concede that this does break many legacy maps in a lot of ways. 

so yes, this needs discussion.
